code/modules/games/cards.dm

```diff
@@ -31,6 +39,7 @@
 				cards += P
 			qdel(H)
 			to_chat(user,"<span class='notice'>You place your cards on the bottom of [src]</span>.")
+			update_icon()
```

Comment:
since you override `update_icon()` to only `return`, I'm not sure calling it will actually do anything here.
```
/obj/item/deck/update_icon()
	return
```